### PR TITLE
Add USE_MKL2017 back.

### PR DESCRIPTION
**Pull request by zheng-da**
## Description ##
This is to add the backward compatibility to compile MKLDNN.

### Essentials ###
- [ ] Passed code style checking (`make lint`)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by marcoabreu** [comment]
@szha could the test result http://jenkins.mxnet-ci.amazon-ml.com/job/incubator-mxnet/job/PR-9650/1/display/redirect be related to the MKLDNN change?

File: Makefile [hunk 1]
```diff
@@ -59,6 +59,10 @@ endif
 # use customized config file
 include $(config)
 
+ifeq ($(USE_MKL2017), 1)
+	USE_MKLDNN=1
```
Comment (marcoabreu) [inline Makefile]: Print a message that we're automatically switching to MKLDNN and that the usage of this option is deprecated

**Comment by szha** [comment]
Yes, I think Da is looking into it.

**Comment by marcoabreu** [comment]
Obsolete as of #9657

**Comment by marcoabreu** [comment]
Obsolete as of https://github.com/apache/incubator-mxnet/pull/9661. Pleas incorporate this change in the new PR.
